src/formatting_stack/impl/overrides.clj

```diff
@@ -0,0 +1,57 @@
+(ns formatting-stack.impl.overrides
+  (:require
+   [clojure.spec.alpha :as spec]
+   [formatting-stack.protocols.spec :as protocols.spec]
+   [formatting-stack.util :refer [rcomp]]
+   [nedap.speced.def :as speced]))
+
+(defn removing-deep-merge
+  "A deep-merge in which `nil` values express that the given key-value entry should be removed."
+  ([a]
+   (if-not (map? a)
+     a
+     (-> (into {}
+               (filter (fn [[k v]]
+                         (some? v)))
+               a)
+         (with-meta (meta a)))))
```

Comment:
```clj
     (->> (remove (comp nil? second) a)
          (into (with-meta {} (meta a))))
```

Reply:
```clj
(-> (into {}
          (filter (comp val some?))
          a)
    (with-meta (meta a)))
```

* prefer `val` to `second`
* Correct `comp` order (it's the opposite in transducers)
* `remove nil?` reads like a double negative. So, `filter some? `
* Conceptually `into` happens first and `with-meta` second. First things should be read earlier (top-to-bottom)